src/NuGet.Core/NuGet.Packaging/Signing/Signatures/AuthorPrimarySignature.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if IS_DESKTOP
+using System.Security.Cryptography.Pkcs;
+#endif
+
+namespace NuGet.Packaging.Signing
+{
+    public class AuthorPrimarySignature : PrimarySignature
+    {
+#if IS_DESKTOP
+
+        public AuthorPrimarySignature(SignedCms signedCms)
+            : base(signedCms, SignatureType.Author)
+        {
+            VerifySigningTimeAttribute(SignerInfo);
+        }
+
+        private static void VerifySigningTimeAttribute(SignerInfo signerInfo)
```

Comment:
This verification applies to [author signatures](https://github.com/NuGet/Home/wiki/Package-Signatures-Technical-Details#-author-signatures), [repository signatures and repository countersignatures](https://github.com/NuGet/Home/wiki/Repository-Signatures-and-Countersignatures-Technical-Specification#repository-signatures-and-countersignatures).